EngineLayer/Proteomics/PeptideWithSetModifications.cs

```diff
@@ -55,16 +55,13 @@ internal PeptideWithSetModifications(PeptideWithSetModifications modsFromThisOne
 
         #region Public Properties
 
-        public CompactPeptide CompactPeptide
+        public CompactPeptide CompactPeptide(TerminusType terminusType)
         {
-            get
+            if (compactPeptide == null)
```

Comment:
The way it is now, is if you ask for one terminus type and then another, it will return the first one. I think it should be a dictionary